homepage/models.py

```diff
@@ -67,10 +185,54 @@ def get_elective_courses(courses):
         # gets all courses that are electives
         return courses.filter(mandatory=False)
 
-    # --- returns if course has Prerequisites
-    def has_preqs(self):
-        return Prerequisites.does_course_have_prerequisites(self)
+    @staticmethod
+    def get_courses_with_preqs(courses):
+        # gets all courses that have prerequisites
+        have_preqs = []
+        for course in courses:
+            if course.has_preqs():
+                have_preqs.append(course.course_id)
+        return courses.filter(course_id__in=have_preqs)
+
+    @staticmethod
+    def get_courses_without_preqs(courses):
+        # gets all courses that have prerequisites
+        no_preqs = []
+        for course in courses:
+            if not course.has_preqs():
+                no_preqs.append(course.course_id)
+        return courses.filter(course_id__in=no_preqs)

```

Comment:
If you move lines 191-194 into a new function ```get_courses_with_preqs_list```,
then it will be possible to call from this function, WDYT?
```suggestion
        have_preqs = get_courses_with_preqs_list(courses)
        return courses.exclude(course_id__in=have_preqs)
```

Reply:
I'm confused as to why you put this at lines 201-204, but other than that, I could indeed do that.

Reply:
If there is going to be a function ```get_courses_with_preqs_list```, so both ```get_courses_without_preqs```
and ```get_courses_with_preqs``` can be shortend.
First call the new function ```get_courses_with_preqs_list``` and then ```courses.exclude``` or ```courses.filter``` depends on the function, anyway it's optional.
Anyway, it's optional.